app/sequencescape_excel/sequencescape_excel/helpers/worksheet.rb

```diff
@@ -44,10 +49,21 @@ def styles
         @styles ||= {}
       end
 
+      def find_or_create_style(components)
+        styles[components] ||= generate_style(components)
+      end
+
       def create_styles
         styles.tap { |s| STYLES.each { |k, style| s[k] = Style.new(workbook, style) } }
       end
 
+      private
+
+      def generate_style(components)
+        style = components.reduce({}) { |store, component| store.merge!(STYLES.fetch(component, {})) }
+        Style.new(workbook, style)
+      end
```

Comment:
Build a style by merging all those specified.

We ignore unrecognised styles here so as to let new types (and even locked) fall back to the default styles.